### PR TITLE
feat: Enable registration based on mentors availability

### PR DESCRIPTION
**Pull request by dricazenck**
## Description

For ad-hoc sessions on a monthly basis, it is necessary to enable mentors based on their availability. 

Given that mentors may have availability in specific months, it would be more flexible to establish a structured timetable. This would involve reaching out to mentors beforehand to ascertain their upcoming availability.

Technical details:

Additional fields in yml should look like (we can put only two months for now and later extend):

```
availability: [5, 6, 7, 8] 
```

Logic is following:

Button should be enabled the first 7 days of the month and disabled after that.

## Change Type
- [X] New Feature

## Related Issue
https://github.com/WomenWhoCode/london/issues/129

## Screenshots

image

In case it is not in the first 7 days in the month, the page will looks like this:
image

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] I checked and followed the [contributor guide](CONTRIBUTING.md) 
- [x] I have tested my changes locally.
- [x] I have added a screenshot from the website after I tested it locally

### CODE REVIEW

File: _data/mentors.yml [hunk 1]
```diff
@@ -78,9 +74,7 @@
   type: both
   index: 3
   languages: Portuguese, English
-  availability:
-    september: false
-    october: false
+  availability: [5]
```
Comment (dricazenck) [inline _data/mentors.yml:77]:
```suggestion
  availability: [ 6, 7, 8 ]
```

**Review by mads6879 — APPROVED** [review]
The core logic in registration.html looks good to me. 
Rest of the changes are for maintainability of code (logic for networks, review and resources moved to separate files).

**Review by nora-weisser — APPROVED** [review]
Implementation and logic looks good to me! 🚀💯

**Comment by dricazenck** [comment]
> Rest of the changes are for maintainability of code (logic for networks, review and resources moved to separate files).

I based this changes in the previous pull request, but now that is updated you can see only related changes about ad-hoc registration.

**Pull request merged**